lib/Sema/ConstraintSystem.cpp

```diff
@@ -1711,6 +1713,14 @@ ConstraintSystem::getTypeOfReference(ValueDecl *value,
     auto numLabelsToRemove = getNumRemovedArgumentLabels(
         funcDecl, /*isCurriedInstanceReference=*/false, functionRefKind);
 
+    if (Context.LangOpts.hasFeature(Feature::InferSendableFromCaptures)) {
+      // All global functions should be @Sendable
+      if (!funcDecl->getDeclContext()->isTypeContext()) {
```

Comment:
I thought I left a comment about this but I cannot find it. The check as it is right now doesn't preclude "local" function declarations only methods on types (including protocols). 

Do we actually want local function to be include here? I'd think not because they can capture values from context right?

Reply:
That's right, local functions are only `@Sendable` when they are explicitly annotated.

Reply:
Right, I'll add a test case for this 

```
func outerFunc() {
  func innerFunc() {}
  
  let q : @Sendable () -> Void = innerFunc // error
}
```

Reply:
done  :)